File: internal/content/testdata/bench/typescript/best-time-to-buy-stock.ts
function max_profit(prices: number[]): number {
    let minPrice = prices[0];
    let profit = 0;
    for (let i = 1; i < prices.length; i++) {
        profit = Math.max(profit, prices[i] - minPrice);
        minPrice = Math.min(minPrice, prices[i]);
    }
    return profit;
}

File: internal/content/testdata/bench/java/best-time-to-buy-stock.java
import java.util.*;

class Solution {
    public long max_profit(long[] prices) {
        long minPrice = prices[0];
        long profit = 0;
        for (int i = 1; i < prices.length; i++) {
            profit = Math.max(profit, prices[i] - minPrice);
            minPrice = Math.min(minPrice, prices[i]);
        }
        return profit;
    }
}
